### PR TITLE
Fqdn: log misbehaving applications that do not respect DNS TTL

### LINKED ISSUE #14320: Identify misbehaving applications that do not respect DNS TTL
A user report of a bug came from apps not respecting the DNS TTL which led to the following fix: https://github.com/cilium/cilium/pull/13914. This fix is an example of Cilium doing a best-effort attempt to keep connectivity for these apps. To improve troubleshooting, it can be helpful to identify when applications are misbehaving, so that the appropriate steps can be taken. A log msg can be sufficient as an indication.

### PR DESCRIPTION
**Pull request by youssefazrak**
A user report of a bug came from apps not respecting the DNS TTL which
led to the following fix: #13914.
To improve troubleshooting, this PR introduces a log message when an
application is misbehaving.

Fixes: #14320

Signed-off-by: Youssef Azrak <yazrak.tech@gmail.com>

### CODE REVIEW

File: pkg/fqdn/cache.go [hunk 1]
```diff
@@ -789,6 +790,10 @@ func (zombies *DNSZombieMappings) isZombieAlive(zombie *DNSZombieMapping, aliveN
 
 	for _, name := range zombie.Names {
 		if _, ok := aliveNames[name]; ok {
+			log.WithFields(logrus.Fields{
+				"FQDN": name,
+				"IP":   zombie.IP,
```
Comment (sayboras) [inline pkg/fqdn/cache.go]: nit: IP -> logfields.IPAddr, FQDN -> logfields.DNSName

Comment (youssefazrak) [reply pkg/fqdn/cache.go]: Thanks Tam, fixed now.

File: pkg/fqdn/cache.go [hunk 2]
```diff
@@ -25,6 +25,8 @@ import (
 	"github.com/cilium/cilium/pkg/fqdn/matchpattern"
 	"github.com/cilium/cilium/pkg/ip"
 	"github.com/cilium/cilium/pkg/lock"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+	"github.com/sirupsen/logrus"
```
Comment (sayboras) [inline pkg/fqdn/cache.go:30]: nit: I think cilium project is following the below import convention (i.e one extra line in between)

"std lib"

"cilium"

"thirdparty"

**Review by sayboras — APPROVED** [review]
LGTM :100:

File: pkg/fqdn/cache.go (hunk 2 above)
Comment (youssefazrak) [reply pkg/fqdn/cache.go:30]: Makes sense, fixed now.

**Comment by nathanjsweet** [comment]
test-me-please

**Comment by jibi** [comment]
retest-netnext

**Comment by jibi** [comment]
retest-netnext

**Review by aanm — APPROVED** [review]

**Review by christarazi — COMMENTED** [review]
LGTM, thanks for the PR. (I had this on my queue and forgotten about it)

**Pull request merged**